### DISCUSSION TITLE
Post-processing plt files

### DISCUSSION THREAD

**Discussion by bazharz** [AMReX-Combustion/PeleLMeX#241] [Q&A]
I would like to perform post-processing on plt files, but I am having trouble understanding the data storage format within these files. 
Specifically, I want to extract the temperature field and calculate the temperature gradient using Python. Is it possible to achieve this?

**Comment by esclapez** [comment]
You can do a lot of things with Python using [yt](https://yt-project.org/doc/examining/loading_data.html). There are examples of yt scripts reading Pele pltfile in here:
https://github.com/nickwimer/ytScripts

Note however that computing gradients on multi-level AMR grids is not straightforward because pltfiles do not contain ghost cells, such that stencil-based operations will have issues at coarse/fine boundaries. Pele provides a set of C++ analysis tools [PeleAnalysis](https://github.com/AMReX-Combustion/PeleAnalysis) that use the AMReX machinery to add and fill ghost cells in order to do this type of operations when needed. In particular the `grad` tool can compute the gradient for you and you can then open the resulting file with yt to access the data in python.
